src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -176,6 +176,225 @@ void ImageReplayer<I>::BootstrapProgressContext::update_progress(
 }
 
 template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::send(Context *on_finish) {
+  dout(20) << dendl;
+
+  if (m_on_finish) {
+    dout(20) << "previous request is still in progress, ignoring" << dendl;
+    on_finish->complete(-EAGAIN);
+    return;
+  }
+
+  m_on_finish = on_finish;
+  get_registered_clients();
+}
+
+template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::get_registered_clients() {
```

Comment:
On Tue, Apr 26, 2016 at 1:38 AM, Mykola Golub <notifications@github.com>
wrote:

> I will use the cached clients, though there is no idea if negative
> entries_behind_master can be fixed/improved somehow.
>

Really?  rbd-mirror flushes the commit position after each entry is applied
or is that just a side-effect of the test script constantly flushing it?
We'll always have to deal with the possibility that the master hasn't
committed the update.  Just should guard for that possibility within the
status output (e.g. m_entries_behind_master=0 instead of
m_entries_behind_master=-612).


-- 
Jason